lapis/src/main/kotlin/org/genspectrum/lapis/model/AdvancedQueryCustomListener.kt

```diff
@@ -0,0 +1,534 @@
+                }
+            }
+
+            is SequenceFilterFieldType.FloatFrom -> {
+                throw BadRequestException(
+                    "Cannot use $metadataName in advancedQuery, use >=(${field.name}) instead",
+                    null,
+                )
+            }
+
+            is SequenceFilterFieldType.FloatTo -> {
+                throw BadRequestException(
+                    "Cannot use $metadataName in advancedQuery, use <=(${field.name}) instead",
+                    null,
+                )
+            }
+
+            SequenceFilterFieldType.Int -> {
+                try {
+                    expressionStack.addLast(IntEquals(field.name, metadataValue.toInt()))
+                } catch (e: NumberFormatException) {
+                    throw BadRequestException("'$metadataValue' is not a valid integer", e)
+                }
+            }
+
+            is SequenceFilterFieldType.IntFrom -> {
+                throw BadRequestException(
+                    "Cannot use $metadataName in advancedQuery, use >=(${field.name}) instead",
+                    null,
+                )
+            }
+
+            is SequenceFilterFieldType.IntTo -> {
+                throw BadRequestException(
+                    "Cannot use $metadataName in advancedQuery, use <=(${field.name}) instead",
+                    null,
+                )
+            }
+
+            is SequenceFilterFieldType.StringSearch -> {
+                expressionStack.addLast(StringSearch(field.type.associatedField, metadataValue))
+            }
+
+            SequenceFilterFieldType.VariantQuery -> {
+                throw BadRequestException("$VARIANT_QUERY_FIELD cannot be called from advanced query", null)
+            }
+
+            SequenceFilterFieldType.AdvancedQuery -> {
+                throw BadRequestException("$ADVANCED_QUERY_FIELD cannot be called recursively", null)
+            }
+        }
+    }
+
+    override fun enterRegexMetadataQuery(ctx: AdvancedQueryParser.RegexMetadataQueryContext) {
+        val metadataName = ctx.geneOrName().text
+        val metadataValue = ctx.value().text.trim('\'')
+
+        val field: SequenceFilterField? = allowedSequenceFilterFields.fields[metadataName.lowercase(Locale.US)]
+        field ?: throw BadRequestException("Metadata field $metadataName does not exist", null)
+        when (field.type) {
+            SequenceFilterFieldType.String -> {
+                expressionStack.addLast(StringEquals(field.name, metadataValue))
+            }
+
+            is SequenceFilterFieldType.StringSearch -> {
+                expressionStack.addLast(StringSearch(field.type.associatedField, metadataValue))
+            }
+
+            SequenceFilterFieldType.Lineage -> {
+                val lineage = mapToLineageFilter(
+                    field.name,
+                    metadataValue,
+                )
+                expressionStack.addLast(lineage)
+            }
```

Comment:
IMO this is suprising: This is not a regex. Why hide it in enterRegexMetadataQuery? Does it need the quotes logic?

Reply:
I wanted to keep this option in case we ever have lineages with a space or another special character (apart from *, - or _) that we haven't yet thought about. This way if you enclose a lineage search with ' ' it will still be recognized.

Reply:
What about `country = 'United Kingdom'`? This also has a space and thus needs quotes. Will it be parsed as a regex? The more I think about it, the less it makes sense to distinguish based on quotes around the value. 

Maybe we should always allow quotes around values in the `metadataQuery` while parsing and sort out the quotes later?

Reply:
No it won't parsed as regex because it is country = United Kingdom and country is of type String not Substring - so it will be parsed as an equal

Reply:
So essentially I always allow quotes around strings they are just only required if you have space or a special character as otherwise the parser doesn't work

Reply:
As I mentioned I should not have called this query type regex as it just processes any query that has quotes around the value that the parser identified - I have thus changes the name

Reply:
If you prefer I can merge quotedMetadataQuery and metadataQuery in the parser - this will then allow quotes around all types including floats and ints though. But if we are ok with that it will mean one less function in the CustomListener

Reply:
I'd prefer the one less function in the parser. `age = '5'`  will yield something like a `'5' is not a valid int` exception, doesn't it? IMO that's good enough, users will understand that they should not put quotes around numbers.

Reply:
I imagine that both could work and it's essentially a decision how we want to have it working (trimming `'` or erroring). I have no strong opinion.

Reply:
Done